keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -1047,7 +1058,266 @@ such objects would test that the workload does not crowd out the
+I0821 18:10:34.205935    5468 pathrecorder.go:247] kube-aggregator: "/api/v1/nodes" satisfied by prefix /api/
+I0821 18:10:34.205947    5468 handler.go:143] kube-apiserver: GET "/api/v1/nodes" satisfied by gorestful with webservice /api/v1
+I0821 18:10:34.207719    5468 httplog.go:90] GET /api/v1/nodes: (2.440132ms) 200 [hyperkube/v1.16.0 (linux/amd64) kubernetes/99a8c8c/system:serviceaccount:kube-system:pod-garbage-collector 127.0.0.1:42768]
+```
+
+Generic, on system collection.
+```
+I0821 18:45:46.552359    5468 reqmgmt.go:263] Serving
+  requestInfo=&{false /apis/batch/v1 get apis       []},
+  user=&{system:serviceaccount:kube-system:generic-garbage-collector
+    23af219e-bc94-4814-aa9b-bae1aaa6d954
+    [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated]
+    map[]},
+  fs="kube-controller-manager" after fair queuing
+I0821 18:45:46.552487    5468 handler.go:153] kube-aggregator: GET "/apis/batch/v1" satisfied by nonGoRestful
+I0821 18:45:46.552507    5468 pathrecorder.go:240] kube-aggregator: "/apis/batch/v1" satisfied by exact match
+I0821 18:45:46.552521    5468 handler.go:143] kube-apiserver: GET "/apis/batch/v1" satisfied by gorestful with webservice /apis/batch/v1
+I0821 18:45:46.552642    5468 httplog.go:90] GET /apis/batch/v1?timeout=32s: (398.571µs) 200 [hyperkube/v1.16.0 (linux/amd64) kubernetes/99a8c8c/system:serviceaccount:kube-system:generic-garbage-collector 127.0.0.1:42768]
+```
+
+#### Kube-Scheduler
+
+LIST
+```
+I0821 18:08:48.572112    5468 reqmgmt.go:263] Serving
+  requestInfo=&{true /api/v1/persistentvolumeclaims list api  v1
+    persistentvolumeclaims   [persistentvolumeclaims]},
+  user=&{system:kube-scheduler  [system:authenticated] map[]},
+  fs="kube-scheduler" after fair queuing
+I0821 18:08:48.572211    5468 handler.go:153] kube-aggregator: GET "/api/v1/persistentvolumeclaims" satisfied by nonGoRestful
+I0821 18:08:48.572226    5468 pathrecorder.go:247] kube-aggregator: "/api/v1/persistentvolumeclaims" satisfied by prefix /api/
+I0821 18:08:48.572237    5468 handler.go:143] kube-apiserver: GET "/api/v1/persistentvolumeclaims" satisfied by gorestful with webservice /api/v1
+I0821 18:08:48.572587    5468 httplog.go:90] GET /api/v1/persistentvolumeclaims?limit=500&resourceVersion=0: (904.5µs) 200 [hyperkube/v1.16.0 (linux/amd64) kubernetes/99a8c8c/scheduler 127.0.0.1:42784]
+```
+
+WATCH
+```
+I0821 18:08:48.577984    5468 reqmgmt.go:64] Serving
+  RequestInfo=&request.RequestInfo{IsResourceRequest:true,
+    Path:"/apis/apps/v1/replicasets", Verb:"watch", APIPrefix:"apis",
+    APIGroup:"apps", APIVersion:"v1", Namespace:"",
+    Resource:"replicasets", Subresource:"", Name:"",
+    Parts:[]string{"replicasets"}},
+  user.Info=&user.DefaultInfo{Name:"system:kube-scheduler", UID:"",
+    Groups:[]string{"system:authenticated"},
+    Extra:map[string][]string(nil)}
+  as longrunning
```

Comment:
there are other long running requests, like exec, port-forward, log, ... are they checked?

Reply:
The point of _this_ KEP update is not to prove that the implementation is doing a good job of classification, so do not pay attention to the classification results shown here.  The point of this update is only to provide some input to the process of _designing_ the classification.

The existing implementation is identifying longrunning requests the same way that the max-in-flight filter does.  Do you have reason to doubt that this is good?

Reply:
ok, that makes sense. 

i don't have reason to doubt it, just figured they are very different than WATCH.